woocommerce/woocommerce
# CYS: Fix Header/Footer template parts disappear

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





| Before | After |
|--------|--------|
|<video src=https://github.com/woocommerce/woocommerce/assets/4463174/785a8453-73bc-45f2-a3ad-cacd529bd490 /> | <video src=https://github.com/woocommerce/woocommerce/assets/4463174/d491ec37-e5d7-4c38-945f-8b31d5be34cb /> | 


This PR fixes the Header/Footer template parts disappear, or at least, greatly reduces the chances of this happening.

This issue happens because, for some reason, the `innerBlocks` of the template part block became 0 ([source code](https://github.com/WordPress/gutenberg/blob/0c176fef1896ee5264526601825a81b229d9ddd2/packages/block-library/src/template-part/edit/index.js/#L108-L109)).

This looks like a bug (https://github.com/WordPress/gutenberg/issues/20243) within the code responsible for synchronizing inner blocks on every change: https://github.com/WordPress/gutenberg/blob/0c176fef1896ee5264526601825a81b229d9ddd2/packages/block-editor/src/components/inner-blocks/use-inner-block-template-sync.js/#L56-L113.


In this PR, I removed all the changes we applied to the blocks array, passing the original one returned by the hook. I imagine that some optimizations are checking the reference.
Futhermore, I added a check on `SidebarNavigationScreenHomepage` to trigger the onChange only when the pattern is changed. 


Closes #45711 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have the WooCommerce Beta Tester plugin installed and activated.
2. Under Tools > WCA Test Helper > Features, make sure customize-store is enabled.
3. Head over to WooCommerce > Home > Customize your store.
4. Click on "Start designing" and proceed to the Pattern Assembler.
5. Click on "Design your homepage".
6. Switch the various patterns very quickly (check the video).
7. Ensure that the header and the footer don't disappear.
8. Go back.
9. Click on "Choose your Header".
10. Change it and ensure that the header is highlighted.
11. Go back.
12. Click on "Choose your Footer".
13. Change it and ensure that the header is highlighted.
14. Go Back.
15. Click done.
16. Click view store.
17. Ensure that all the changes are visible on the frontend



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

CYS: Fix Header/Footer template parts disappear

#### Comment 

</details>
